app/main/views/applications.py

```diff
@@ -166,3 +168,9 @@ def revert_application():
 def search_applications(keyword):
     result = data_api_client.req.applications().search(keyword).get()
     return jsonify(result)
+
+
+@main.route('/applications/<int:id>', methods=['DELETE'])
```

Comment:
I usually add a route where the id is optional so you don't need to specify one in url_for